[GitHub Workflows] Prime Renv Cache canot initialise {renv} project after version 1.0.0

During the Prime Renv Cache step in the R CMD Check workflow, we set up and then tear down a {renv} project in order to prime the cache with the packages we need to build lessons. 

https://github.com/carpentries/sandpaper/blob/8030c16ca50d30903af15097f082d03a6fe1eef7/.github/workflows/R-CMD-check.yaml#L101-L108

With version 1.0.0, we are getting a new error:

```r
Not interactive. Will:
Use only the DESCRIPTION file. (explicit mode)
- Using '1' snapshot type. Please see `?renv::snapshot` for more details.

Error: Error: internal error: unhandled snapshot type '1' in renv_snapshot_dependencies_impl(project, type, dev)
Execution halted
Traceback (most recent calls last):
8: renv::init()
7: deps <- renv_snapshot_dependencies(project, type = type, dev = TRUE) at init.R#128
6: packages <- dynamic(
     list(project = project, type = type, dev = dev),
     renv_snapshot_dependencies_impl(project, type, dev)
   ) at snapshot.R#945
5: the$dynamic_objects[[id]] <- the$dynamic_objects[[id]] %||% {
     dlog("dynamic", "memoizing dynamic value for '%s'", id)
     value
   } at dynamic.R#58
4: packages <- dynamic(
     list(project = project, type = type, dev = dev),
     renv_snapshot_dependencies_impl(project, type, dev)
   ) at snapshot.R#945
3: path <- case(
     type %in% c("packrat", "implicit") ~ project,
     type %in% "explicit" ~ file.path(project, "DESCRIPTION"),
     ~ {
       fmt <- "internal error: unhandled snapshot type '%s' in %s"
       stopf(fmt, type, stringify(sys.call()))
     }
   ) at snapshot.R#969
2: stopf(fmt, type, stringify(sys.call()))
1: stop(sprintf(fmt, ...), call. = call.) at utils-format.R#3
Error: Process completed with exit code 1.
```

--- a/.github/workflows/R-CMD-check.yaml
+++ b/.github/workflows/R-CMD-check.yaml
@@ -11,6 +11,7 @@ on:
     branches: [main, master]
   schedule:
     - cron: '0 0 * * 2'
+  workflow_dispatch:
 
 name: R-CMD-check
 
@@ -101,6 +102,7 @@ jobs:
       - name: "Prime {renv} Cache"
         if: runner.os != 'Windows'
         run: |
+          renv::settings$snapshot.type("explicit")
           renv::init()
           system('sudo rm -rf renv.lock renv .Rprofile')
           system('git clean -fd -e .github')

--- a/.github/workflows/test-coverage.yaml
+++ b/.github/workflows/test-coverage.yaml
@@ -5,6 +5,7 @@ on:
     branches: [main, master]
   pull_request:
     branches: [main, master]
+  workflow_dispatch:
 
 name: test-coverage
 
@@ -68,6 +69,7 @@ jobs:
 
       - name: Prime {renv} Cache
         run: |
+          renv::settings$snapshot.type("explicit")
           renv::init()
           system('rm -rf renv .Rprofile')
           system('git clean -fd -e .github')

--- a/tests/testthat/test-manage_deps.R
+++ b/tests/testthat/test-manage_deps.R
@@ -201,12 +201,8 @@ test_that("update_cache() will update old package versions", {
   skip_if_offline()
   skip_if(covr::in_covr())
 
-  withr::local_options(list("renv.verbose" = TRUE))
 
-  suppressMessages({
-    res <- update_cache(path = fs::path(lsn, "episodes"), prompt = FALSE, quiet = FALSE) %>%
-      expect_output("sessioninfo")
-  })
+  res <- update_cache(path = fs::path(lsn, "episodes"), prompt = FALSE, quiet = FALSE)
   expect_true(
     package_version(res$sessioninfo$Version) > package_version("1.1.0")
   )

